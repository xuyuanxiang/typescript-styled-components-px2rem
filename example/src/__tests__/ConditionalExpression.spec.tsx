import TestUtils from 'react-dom/test-utils';
import ReactDOM from 'react-dom';
import { ThemeProvider } from 'styled-components';
import { ConditionalExpression } from '../ConditionalExpression';
import React from 'react';

describe('embedded ConditionalExpression', () => {
  const div: HTMLDivElement = document.createElement('div');

  beforeEach(() => {
    document.body.appendChild(div);
  });

  afterEach(() => {
    ReactDOM.unmountComponentAtNode(div);
    document.body.removeChild(div);
  });

  it("should transform <ConditionalExpression/> with ThemeProvider's fontSize", function() {
    TestUtils.act(() => {
      ReactDOM.render(
        <ThemeProvider theme={{ fontSize: 48 }}>
          <ConditionalExpression />
        </ThemeProvider>,
        div,
      );
    });
    const button = div.querySelector('button');
    if (button) {
      expect(document.documentElement).toMatchSnapshot();
      const style = getComputedStyle(button);
      expect(style.fontSize).toBe('0.48rem');
    } else {
      throw new Error('ConditionalExpression should be render');
    }
  });

  it('should transform <ConditionalExpression/> with self fontSize', function() {
    TestUtils.act(() => {
      ReactDOM.render(
        <ThemeProvider theme={{ fontSize: 48 }}>
          <ConditionalExpression fontSize={24} />
        </ThemeProvider>,
        div,
      );
    });
    const button = div.querySelector('button');
    if (button) {
      expect(document.documentElement).toMatchSnapshot();
      const style = getComputedStyle(button);
      expect(style.fontSize).toBe('0.24rem');
    } else {
      throw new Error('ConditionalExpression should be render');
    }
  });
});
