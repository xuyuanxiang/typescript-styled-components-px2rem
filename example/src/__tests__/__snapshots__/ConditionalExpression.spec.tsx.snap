// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`embedded ConditionalExpression should transform <ConditionalExpression/> with ThemeProvider's fontSize 1`] = `
<html>
  <head>
    <style
      data-styled=""
      data-styled-version="4.4.1"
    >
      
      
/* sc-component-id: sc-bdVaJa */
.iTFtex{font-size:0.48rem;}
    </style>
  </head>
  <body>
    <div>
      <button
        class="sc-bdVaJa iTFtex"
      />
    </div>
  </body>
</html>
`;

exports[`embedded ConditionalExpression should transform <ConditionalExpression/> with self fontSize 1`] = `
<html>
  <head>
    <style
      data-styled=""
      data-styled-version="4.4.1"
    >
      
      
/* sc-component-id: sc-bdVaJa */
.iTFtex{font-size:0.48rem;}
      
/* sc-component-id: sc-bwzfXH */
.eSDCcM{font-size:0.24rem;}
    </style>
  </head>
  <body>
    <div>
      <button
        class="sc-bwzfXH eSDCcM"
      />
    </div>
  </body>
</html>
`;
